Framework/LiveData/src/ISIS/ISISHistoDataListener.cpp

```diff
@@ -196,16 +208,9 @@ boost::shared_ptr<Workspace> ISISHistoDataListener::extractData() {
 
   // Create the 2D workspace for the output
   auto localWorkspace = WorkspaceFactory::Instance().create(
-      "Workspace2D", numberOfHistograms, m_numberOfBins[m_timeRegime] + 1,
+      bufferWorkspace, numberOfHistograms, m_numberOfBins[m_timeRegime] + 1,
       m_numberOfBins[m_timeRegime]);
 
-  // Set the unit on the workspace to TOF
-  localWorkspace->getAxis(0)->unit() =
-      Kernel::UnitFactory::Instance().create("TOF");
-  localWorkspace->setYUnit("Counts");
-  localWorkspace->updateSpectraUsing(
-      SpectrumDetectorMapping(m_specIDs, m_detIDs));
-
```

Comment:
2 lines below there is a call to `calculateIndicesForReading`. Can you also move that into `connect()` above?

Also, have a look below at the block starting with `if (period > firstPeriod) {`. Can the extra call to `create` be removed now?